Assignment 5.Rmd

```diff
@@ -1,82 +1,74 @@
-
 library(rpart)
 library(party)
 ```
 
 ## Part I
 ```{r}
-D1 <- 
+D1 <- read.table("intelligent_tutor.csv", sep = ",", header = TRUE)
 ```
 
 ##Classification Tree
 First we will build a classification tree to predict which students ask a teacher for help, which start a new session, or which give up, based on whether or not the student completed a session (D1$complete) and whether or not they asked for hints (D1$hint.y). 
 ```{r}
-
 c.tree <- rpart(action ~ hint.y + complete, method="class", data=D1) #Notice the standard R notion for a formula X ~ Y
-
 #Look at the error of this tree
 printcp(c.tree)
-
+#CP stands for Complexity Parameter - this value represents the cost in complexity for splitting a node. It reflects a ration between the number of nodes and the accuracy of the model. The default cutoff is 0.010.
+#nsplit stands for the number of splits in the tree
+#error stands for the overall percent error in predictions of the tree
+#xerror stands for the average cross-validated error in the tree
+#xstd stands for the average cross-validated standard devation of the error in teh tree
 #Plot the tree
 post(c.tree, file = "tree.ps", title = "Session Completion Action: 1 - Ask teacher, 2 - Start new session, 3 - Give up")
-
+#This plot creates a new file - a postscript file that is like a pdf
 ```
 ## Part II
 
 #Regression Tree
 
 We want to see if we can build a decision tree to help teachers decide which students to follow up with, based on students' performance in Assistments. We will create three groups ("teacher should intervene", "teacher should monitor student progress" and "no action") based on students' previous use of the system and how many hints they use. To do this we will be building a decision tree using the "party" package. The party package builds decision trees based on a set of statistical stopping rules.
 
-#Visualize our outcome variable "score"
+#Take a look at our outcome variable "score"
 ```{r}
-
+hist(D1$score)
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
 ```{r}
-D1$advice <- 
+D1$advice <- ifelse(D1$score <=0.4, "intervene", ifelse(D1$score > 0.4 & D1$score <=0.8, "monitor", "no action"))
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+score_ctree <- ctree(factor(advice) ~ prior_prob_count + prior_percent_correct + hints, D1)
 ```
 
 #Plot tree
 ```{r}
-
+plot(score_ctree)
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+There are several ways you could answer this question. The main point woul dbe to justify your answer by referring to the distribution of students in each terminal node. For example, students who ask for more than 12 hints are very likely to need intervention.
```

Comment:
What is the second behavior that the teacher should most closely pay attention to?